### ISSUE TITLE
UTF-8 not supported in resource_formats.json

### ISSUE THREAD

**Issue by hecht-software** [ckan/ckan#6054]
**CKAN version**

CKAN 2.9.2
Ubuntu 18.04
Python 3.6

**Describe the bug**
When trying to upload a resource I get a message "Internal server error".

The log file reveals an error that occurs when resource_formats.json is loaded:
`ValueError: Invalid JSON syntax in /usr/lib/ckan/default/src/ckan/ckan/config/resource_formats.json: 'ascii' codec can't decode byte 0xc2 in position 4411: ordinal not in range(128)`.

The problem seems to be caused by utf-8 characters in the JSON file.

**Steps to reproduce**
- Add utf-8 characters to resource_formats.json
- Upload resource

**Expected behavior**
The resource is uploaded without error log

**Additional details**
It seems that JSON files are not loaded as utf-8 by default and that you have to specify the encoding (according to https://stackoverflow.com/a/35052042/395879).

The relevant code is probably `ckan/lib/helpers.py` in line 2685:
```python
        with open(format_file_path) as format_file:
            try:
                file_resource_formats = json.loads(format_file.read())
```

Stack trace:

```
  File "/usr/lib/ckan/default/src/ckan/ckan/logic/action/update.py", line 295, in package_update
    package_plugin, context, data_dict, schema, 'package_update')
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/plugins.py", line 306, in plugin_validate
    return toolkit.navl_validate(data_dict, schema, context)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/navl/dictization_functions.py", line 273, in validate
    converted_data, errors = _validate(flattened, schema, validators_context)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/navl/dictization_functions.py", line 314, in _validate
    convert(converter, key, converted_data, errors, context)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/navl/dictization_functions.py", line 224, in convert
    value = converter(converted_data.get(key))
  File "/usr/lib/ckan/default/src/ckan/ckan/logic/validators.py", line 798, in clean_format
    return h.unified_resource_format(format)
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/helpers.py", line 2713, in unified_resource_format
    formats = resource_formats()
  File "/usr/lib/ckan/default/src/ckan/ckan/lib/helpers.py", line 2691, in resource_formats
    (format_file_path, e))
ValueError: Invalid JSON syntax in /usr/lib/ckan/default/src/ckan/ckan/config/resource_formats.json: 'ascii' codec can't decode byte 0xc2 in position 4411: ordinal not in range(128)
```

**Comment by Gauravp-NEC** [comment]
@hecht-software , You have committed the fix in your own master branch. It will not get review/merge unless and until you commit this fix on CKAN master branch.

**Comment by BonaBeavis** [comment]
Why is this issue closed? It still exists with  ckan 2.9.5 and py3.
